### PR TITLE
[TD-883]: Move all of the ci specific bits to ci/ path.

### PR DESCRIPTION
**Pull request by asutosh**
## Description
To facilitate automatic sync between all the release branches in the event of any change in tests, or ci specific stuff.
Moving anything that comes under the purview of ci to a separate directory ci/ makes this easier.

## Motivation and Context
Whenever changes in tests, or ci related files are made, it has to be propagated across
multiple release branches. To make this automated through automated syncing across branches and PRs, we
need to set easy boundaries where the CI specific files reside. Moving everything to a `ci/` folder makes it a bit easier to
implement this.

## How This Has Been Tested

As part of the automated checks as part of this PR, and also a release flow which will be initiated by pushing a tag.

### CODE REVIEW

**Comment by Tyk-ITS** [comment]
**API tests result: skipped** :no_entry_sign: 
Branch used: refs/pull/3892/merge
Commit:  
Triggered by: pull_request (@asutosh)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/1853297891)

**Comment by Tyk-ITS** [comment]
**API tests result: skipped** :no_entry_sign: 
Branch used: refs/pull/3892/merge
Commit: 09334ec39f6ba60080efc0a154b451657d5db5e7 
Triggered by: pull_request (@asutosh)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/1853500317)

File: .github/workflows/ci-tests.yml [hunk 1]
```diff
@@ -71,7 +71,7 @@ jobs:
       - name: Run Gateway Tests
         id: ci-tests
         run: |
-          ./bin/ci-tests.sh 2>&1 | tee test.log
+          ./ci/bin/ci-tests.sh 2>&1 | tee test.log
```
Comment (alephnull) [inline .github/workflows/ci-tests.yml:74]: This file will need to be left out for now. This file is owned by the squads.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -380,7 +380,10 @@ jobs:
         run: |
           set -eaxo pipefail
           if [ ! -d integration/smoke-tests ]; then
-             echo "::warning No smoke tests defined"
+             echo "::warning No squad specific smoke tests defined"
```
Comment (alephnull) [inline .github/workflows/release.yml:383]: Let's change this to "No repo specific smoke tests..."

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -392,6 +395,15 @@ jobs:
                   cd -
               fi
           done
+          for d in ci/tests/*/
```
Comment (alephnull) [inline .github/workflows/release.yml:398]: `ci/tests/` to run before the repo specific ones as a failure there would not run these tests.

**Review by alephnull — CHANGES_REQUESTED** [review]
This looks quite neat.

Now that this is working, we will have to make the appropriate changes in `pr.zsh` and run the new templates on this branch.

**Comment by Tyk-ITS** [comment]
**API tests result: skipped** :no_entry_sign: 
Branch used: refs/pull/3892/merge
Commit: fa38be128637fa2cfd4cd45561ebe1ab1601e1f6 
Triggered by: pull_request (@alephnull)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/1857054714)

**Comment by alephnull** [comment]
I think it makes sense to:
- update sync-automation.yml.m4 with new paths
- meta.zsh to create a PR for sync-automation, review and merge that PR
- merge this PR to master, allowing the sync to run

File: .github/workflows/ci-tests.yml (hunk 1 above)
Comment (asutosh) [reply .github/workflows/ci-tests.yml:74]: Then will keep a `bin/` directory in the root, with this file there, and the rest will be kept in `ci/bin`

**Comment by Tyk-ITS** [comment]
**API tests result: skipped** :no_entry_sign: 
Branch used: refs/pull/3892/merge
Commit: 3fe258e348e630a206aeb0a95a2d9f8789f1983e 
Triggered by: pull_request (@asutosh)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/1857186423)

**Comment by Tyk-ITS** [comment]
**API tests result: skipped** :no_entry_sign: 
Branch used: refs/pull/3892/merge
Commit: 4b3ae9fda23f352fe09bade79c2d08ef4859c802 
Triggered by: pull_request (@asutosh)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/1857203427)

**Comment by Tyk-ITS** [comment]
**API tests result: skipped** :no_entry_sign: 
Branch used: refs/pull/3892/merge
Commit: f4c349e20f4d1da9bc98cf5491dc13fd6b44c7f4 
Triggered by: pull_request (@asutosh)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/1857273306)

**Comment by Tyk-ITS** [comment]
**API tests result: skipped** :no_entry_sign: 
Branch used: refs/pull/3892/merge
Commit: b166771e855f7d3d0e7a38ea7fd2e8c404fb548a 
Triggered by: pull_request (@asutosh)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/1857291689)

**Comment by Tyk-ITS** [comment]
**API tests result: skipped** :no_entry_sign: 
Branch used: refs/pull/3892/merge
Commit: 5c93e8aa2b4a22a1b1cdea3e1cade4ecc4629bd4 
Triggered by: pull_request (@asutosh)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/1858395926)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/3892/merge
Commit: e35c1daae5c24d3fd0b789af40e563b5f5f77049 
Triggered by: pull_request (@asutosh)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/1889254373)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/3892/merge
Commit: 5bd9815943e4a0039daffccac81c3b2121a5a608 
Triggered by: pull_request (@asutosh)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/1933580457)

**Review by alephnull — APPROVED** [review]
:shipit:

**Comment by Tyk-ITS** [comment]
**API tests result: failure** :no_entry_sign: 
Branch used: refs/pull/3892/merge
Commit: 9dfe7539a8e59fb7d43495a3ccd5d3f500d0dabc 
Triggered by: pull_request (@asutosh)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/1945004933)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/heads/master
Commit: 4a12f0c1b3810dbef104bad6c5c94096ec41eabc [TD-883]: Move all of the ci specific files to ci/ path. (#3892)

Facilitates automatic sync of ci files to all the release
branches in the event of any change in tests, or any of the  ci specific bits.
This moves anything that comes under the purview of ci to a
separate directory ci/ that makes this process  easier.
Post merging of this PR, sync automation should run and copy over
all the relevant changes to the specified release branches.

[changelog]
internal: Consolidating all ci related files under ci/ path.

* Update paths in goreleaser config files.

* Remove unwanted extra_files from goreleaser.

* Update go.mod for plugin tests.

* Fix install script path on packages.

* Fix path for repo specific smoke-tests.

* Change the order of tests to run ci
  tests first.

* Move back tests maintained by squads to bin/

* Update path for ci tests maintained by squads.

* Add ci/ and smoke-tests/ to sonarcloud exclusions,
since these contain only tests, and not actual code.

* Reverting a mistaken branch rename.

* Sync from latest releng templates

* Sync from latest releng templates
Triggered by: push (@asutosh)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/1945155558)

**Pull request merged**